Pcap++/header/PcapFilter.h

```diff
@@ -484,82 +484,117 @@ namespace pcpp
+		/**
+		 * Remove all filters from the composite filter.
+		 */
+		void clearAllFilters() { m_FilterList.clear(); }
 	};
 
-
-
 	/**
-	 * @class OrFilter
-	 * A class for connecting several filters into one filter with logical "or" between them. For example: if the 2 filters are: "IPv4 address =
-	 * x.x.x.x" + "TCP port dst = 80", then the new filter will be: "IPv4 address = x.x.x.x _OR_ TCP port dst = 80"<BR>
-	 * This class follows the composite design pattern<BR>
-	 * For deeper understanding of the filter concept please refer to PcapFilter.h
-	 * @todo add some methods: "addFilter", "removeFilter", "clearAllFilter"
+	 * Supported composite logic filter operators enum
 	 */
-	class OrFilter : public GeneralFilter
+	enum class CompositeLogicFilterOp
 	{
-	private:
-		std::vector<GeneralFilter*> m_FilterList;
-	public:
-
-		/**
-		 * An empty constructor for this class. Use addFilter() to add filters to the or condition
-		 */
-		OrFilter() {}
+		/** Logical AND operation */
+		AND,
+		/** Logical OR operation */
+		OR,
+	};
 
+	namespace detail
+	{
+		/* Could potentially be moved into CompositeLogicFilter as a private member function, with if constexpr when C++17 is the minimum supported standard.*/
 		/**
-		 * A constructor that gets a list of pointers to filters and creates one filter from all filters with logical "or" between them
-		 * @param[in] filters The list of pointers to filters
+		 * Returns the delimiter for joining filter strings for the composite logic filter operation.
+		 * @return A string li
 		 */
-		explicit OrFilter(std::vector<GeneralFilter*>& filters);
+		template <CompositeLogicFilterOp op> constexpr const char *getCompositeLogicOpDelimiter() = delete;
+		template <> constexpr const char *getCompositeLogicOpDelimiter<CompositeLogicFilterOp::AND>() { return " and "; };
+		template <> constexpr const char *getCompositeLogicOpDelimiter<CompositeLogicFilterOp::OR>() { return " or "; };
+	}
 
-		/**
-		 * Add filter to the or condition
-		 * @param[in] filter The filter to add
-		 */
-		void addFilter(GeneralFilter* filter) { m_FilterList.push_back(filter); }
+	/**
+	 * @class CompositeLogicFilter
+	 * A class for connecting several filters into one filter with logical operation between them.<BR>
+	 * For deeper understanding of the filter concept please refer to PcapFilter.h
+	 */
+	template <CompositeLogicFilterOp op>
+	class CompositeLogicFilter : public CompositeFilter
+	{
+	public:
+		using CompositeFilter::CompositeFilter;
 
-		void parseToString(std::string& result);
+		void parseToString(std::string& result)
+		{
+			result.clear();
+			for (auto it = m_FilterList.cbegin(); it != m_FilterList.cend(); ++it)
+			{
+				std::string innerFilter;
+				(*it)->parseToString(innerFilter);
+				result += '(' + innerFilter + ')';
+				if (m_FilterList.back() != *it)
+				{
+					result += detail::getCompositeLogicOpDelimiter<op>();
+				}
+			}
+		}
 	};
 
+	/**
+	 * @class AndFilter
```

Comment:
not a class, remove @class.

Reply:
question, how does the documentation generator work with aliases?

Reply:
here is  an example
```
	/**
	 * A callback that is called when a packet is captured by PcapLiveDevice
	 * @param[in] packet A pointer to the raw packet
	 * @param[in] device A pointer to the PcapLiveDevice instance
	 * @param[in] userCookie A pointer to the object put by the user when packet capturing stared
	 * @return True when main thread should stop blocking or false otherwise
	 */
	using OnPacketArrivesStopBlocking = std::function<bool(RawPacket*, PcapLiveDevice*, void*)>;
	```

Reply:
Resolved 8ac748b